### PR TITLE
Sanitizes Github, So we can get public access again.

### PR DESCRIPTION
**Pull request by DorainGrey**
## What this does
This removes/replaces all the words, that github flagged as being why our repository is tagged as objectionable content & one or two instances that would definitely get flagged if checked again,
See following list https://docs.google.com/document/d/1IGHR_zjGl_rtitNXuSGkY-TxOudgf9kPpKBAeAUOZps/edit

I did not remove the "Faggot" food, as I think this can be explained in an appeal as it's a real dish.

## Why it's good
The Github being private is terrible for community engagement with the codebase. People should not be forced to create an account just to see the changes being made. 
Anonymity is an important aspect of 4chan culture & should be maintained where possible.
This has virtually no impact whatsoever on the gameplay of the server, you can still say whatever you like in PR's, Comments & on server. 

There is no guarantee Github will accept any appeal based on these changes but TGstation was in a similar position a few years ago and once they removed what was requested they were promptly freed.

### CODE REVIEW

File: code/game/objects/items/devices/PDA/apps/general.dm [hunk 1]
```diff
@@ -142,7 +142,7 @@ var/global/list/currentevents3 = list("Border patrol around Space America has ti
 	"Renowned mime scientist Free Shrugs has discovered a new element today. He has named it '  ', he also says that it has the properties of '   '.",
 	"Archaeologists have discovered god's final message to his creation today. The message reads, 'bawk'.",
 	"Scientists have discovered a new type of elementary particle today. Our sources say it has a bad atitude, and enjoys the color blue.",
-	"Today, a man was discovered to be living with a 20 year old ghost in his house. When the ghost was questioned who killed him, he responded 'A FAGGOT!'. More at four.",
```
Comment (Exxion) [inline code/game/objects/items/devices/PDA/apps/general.dm:145]: What even was the joke here?

File: code/game/objects/items/devices/PDA/apps/general.dm [hunk 2]
```diff
@@ -142,7 +142,7 @@ var/global/list/currentevents3 = list("Border patrol around Space America has ti
...
+	"Today, a man was discovered to be living with a 20 year old ghost in his house. When the ghost was questioned who killed him, he responded 'A Tarjan!'. More at four.",
```
Comment (Exxion) [inline code/game/objects/items/devices/PDA/apps/general.dm]: It's "Tajaran" but I don't even know how this is supposed to be funny

File: code/game/objects/items/devices/PDA/apps/general.dm (hunk 1 above)
Comment (SonixApache) [reply code/game/objects/items/devices/PDA/apps/general.dm:145]: Tajaran*

File: code/game/objects/items/robot/robot_items/robot_harm_alarm.dm [hunk 3]
```diff
@@ -65,7 +65,7 @@
 	desc = "Releases a harmless blast that confuses most organics. For when the crime is JUST TOO MUCH"
 	alarm = "HALT! SECURITY!"
 	alarm_sound = 'sound/voice/halt.ogg'
-	emagged_alarm = "FUCK YOUR CUNT YOU SHIT EATING COCKSUCKER MAN EAT A DONG FUCKING ASS RAMMING SHITFUCK. EAT PENISES IN YOUR FUCKFACE AND SHIT OUT ABORTIONS OF FUCK AND DO A SHIT IN YOUR ASS YOU COCK FUCK SHIT MONKEY FUCK ASS WANKER FROM THE DEPTHS OF SHIT."
+	emagged_alarm = "FUCK YOU Shitcurity,."
```
Comment (SonixApache) [inline code/game/objects/items/robot/robot_items/robot_harm_alarm.dm]: this one should be moved to a patch repo or something
at least make it longer and proper all-caps, like 
```
"FUCK YOU FUCK YOU FUCK YOU FUCK YOU FUCK YOU FUCK YOU FUCK YOU FUCK YOU FUCK YOU FUCK YOU FUCK YOU FUCK YOU FUCK YOU FUCK YOU FUCK YOU FUCK YOU FUCK YOU FUCK YOU FUCK YOU FUCK YOU FUCK YOU FUCK YOU FUCK YOU FUCK YOU FUCK YOU FUCK YOU FUCK YOU."
```

Comment (DorainGrey) [reply code/game/objects/items/robot/robot_items/robot_harm_alarm.dm]: I really don't think Pomf is going to bother to setup a define on the repo frankly but I'll replace it now with the suggested string.

**Comment by Pastamadness** [comment]
Can't we use another site
Is there an actual argument other than laziness

**Comment by DeityLink** [comment]
About fucking time. Thank you for being the adult in the room.

**Comment by Exxion** [comment]
>Can't we use another site
Is there an actual argument other than laziness

They all fucking suck.

**Comment by DorainGrey** [comment]
> Can't we use another site Is there an actual argument other than laziness

Gitlab is not bad, N3xis maintained a mirror there. 
However it's more nuanced than just the code, aside from the pain of everyone needing to recreate accounts. We'd lose a lot of the free CI tools, like the linter, map diff bot etc which other SS13 servers maintain and are github actions.
We'd also lose all the history in terms of PRs etc which is often vital information.

File: code/game/objects/items/devices/PDA/apps/general.dm (hunk 2 above)
Comment (DeityLink) [reply code/game/objects/items/devices/PDA/apps/general.dm]: Yeah in this case a more elaborated joke should be made. The joke was the gratuitous use of the homophobic slur which is very funny for whoever wrote it.

OH WAIT IT WAS KANEF. Fucking lol.

**Comment by DeityLink** [comment]
And that'd be a lot to lose for the sole purpose of having a public code repository with racial slurs etc.

You'd think common sense would prevail in this case but somehow some people need to have it explained to them.

**Comment by Pastamadness** [comment]
Well then since people didn't want this before. I don't see why try it again.

**Comment by DeityLink** [comment]
Because the status quo still doesn't satisfy us. Also I was away when the whole debacle occurred, so this time I get to voice my opinion.

File: code/game/objects/items/devices/PDA/apps/general.dm [hunk 4]
```diff
@@ -142,7 +142,7 @@ var/global/list/currentevents3 = list("Border patrol around Space America has ti
 	"Renowned mime scientist Free Shrugs has discovered a new element today. He has named it '  ', he also says that it has the properties of '   '.",
 	"Archaeologists have discovered god's final message to his creation today. The message reads, 'bawk'.",
 	"Scientists have discovered a new type of elementary particle today. Our sources say it has a bad atitude, and enjoys the color blue.",
-	"Today, a man was discovered to be living with a 20 year old ghost in his house. When the ghost was questioned who killed him, he responded 'A FAGGOT!'. More at four.",
+	"Today, a man was discovered to be living with a 20 year old ghost in his house. When the ghost was questioned who killed him, he responded 'A Tajaran!'. More at four.",
```
Comment (DeityLink) [inline code/game/objects/items/devices/PDA/apps/general.dm]:
```suggestion
	"Today, a man was discovered to be living with a 20 year old ghost in his house. When the ghost was questioned who killed him, he responded 'A BASTARD!'. More at four.",
```
Sorta keeps the vibe a bit more imo. What do you think? Plenty of mainstream games have bastard used as an insult.

**Comment by jwhitak** [comment]
> Well then since people didn't want this before. I don't see why try it again.

Previous PRs tried to do strange things such as skirting around the problem by replacing all references with a define that can be set by servers at runtime, which was dumb. Or just removing all profanity from the entire game (in a PR by adavosk). Those sucked. This one properly removes the references. Surprised that nobody actually did this yet.

**Comment by djkramer123** [comment]
This should've never been up to emojiocracy, GitHub being privated is nothing but harmful to the health of the server, this doesn't effect what you can say in game and anyone still valuing their funny, and mostly unnoticeable, words in the code is just being malicious at this point.

File: code/game/objects/items/devices/PDA/apps/general.dm (hunk 4 above)
Comment (Pastamadness) [reply code/game/objects/items/devices/PDA/apps/general.dm]: This is very problematic and bigoted against people with no fathers.

**Comment by DeityLink** [comment]
Anyone who wishes to use another website is free to go make their own codebase there and open their own server. With Hippie gone we lack a really bad server to act as filter.

File: code/game/objects/items/devices/PDA/apps/general.dm (hunk 4 above)
Comment (DeityLink) [reply code/game/objects/items/devices/PDA/apps/general.dm]: get out of here ada

**Comment by nervere** [comment]
> Well then since people didn't want this before. I don't see why try it again.

The GitHub being open and easily usable, much like code maintainability, is not be a matter of likes/dislikes. It's necessary.

**Comment by Kammerjunk** [comment]
no it isn't
there are legitimate arguments for doing this so use those instead of this overly dramatic nonsense

**Comment by DeityLink** [comment]
> there are legitimate arguments for doing this so use those instead of this overly dramatic nonsense

How about people against this PR provide actual arguments against it instead? Haven't read a single one in this PR so far. Only "bawwww we already tried that" or "can't we just go somewhere else?"

**Comment by Kammerjunk** [comment]
putting aside the absurd idea that you have to argue _against_ something rather than _for_ it, there are absolutely reasons someone might not like this. it does affect the actual game, even if only in minor ways like sanitising the gay religion. they might also have concerns about what the next step is to stay in line with the changing culture of the GitHub staff (the way any group's culture changes). "we already tried this" is _in itself_ a valid objection that can be met in better ways than some offhanded drawsstuff-esque comment about an unknown "us".

on the other hand, there _are_ legitimate arguments for doing this. there's no real evidence that suggests the github being locked has caused us to have fewer coders but because there's not likely to ever be any (how would anyone even measure that?), that's a good argument - not in favour of scrubbing bad words, but at least of being able to access the repo without being logged in.

as a slight aside, this is not the best way to do this and anyone saying it is is pussyfooting around the community that matters (players). no matter how silly it seems, there is a portion of the community that cares about the instances of these words that appear in-game staying. the defines were better

**Comment by Exxion** [comment]
The defines were idiotic and I'd rather just leave the repo private than put that garbage in the code

**Comment by Kammerjunk** [comment]
great, sounds like a solution

**Comment by Exxion** [comment]
Yeah, and so is this
It's ultimately up to Pomf, really

**Comment by DeityLink** [comment]
> in minor ways like sanitising the gay religion.

You'd keep the repo private just so the gay religion may have slurs among its keys? (which don't matter once the religion is initialized)

> they might also have concerns about what the next step is to stay in line with the changing culture of the GitHub staff

ah yes, the good ol' slippery slope fallacy, where have you been my old friend

> "we already tried this" is in itself a valid objection

no it isn't, imagine for a second if people in life never tried something more than once? Here in this particular case, we are an evolving community whose members aren't the same every time. I wasn't there last time for instance. So stop being so disingenuous.

> on the other hand, there are legitimate arguments for doing this.(...)not in favour of scrubbing bad words, but at least of being able to access the repo without being logged in.

That I agree, and to me anonimity should be held at a much higher value by 4channers than racial slurs.

> there is a portion of the community that cares about the instances of these words that appear in-game staying

4 of those changes are in comments that don't even appear in-game. The rest are nothing of value. Get that in your head.

**Comment by Kammerjunk** [comment]
I'd do nothing, I'm telling you the things that get brought up whenever this gets brought up. I don't think you know what the slippery slope fallacy is.

deity, someone objecting to something you say or do isn't the end of the world, it's something you address. "we already tried this so I don't see why do it again" is a valid objection regardless of whether you can respond to it. I'm not going to get anything in my head (it's quite well-ventilated), I just want the discussion about this to be at a slightly higher level than "this is good because I said so," which is the exact sort of comment that lead to all of this.

**Comment by Exxion** [comment]
It is technically possible to include several of these on the secret repo without the use of defines, but who gives a shit
I wouldn't stop someone else from doing it, but I don't think it's worth the trouble, or really worth anything at all

Also, I'm going to point out that the people who benefit most from this PR, players who do not have GitHub accounts, don't get to vote on this or post comments on it. The votes and discussion are entirely from people who stand to gain little to nothing personally from the repo being public again.

As for an actual reason to make the repo public again, I'm going to go with the incredibly obvious "it's not reasonable or fair to expect all players who want to codedive to make a GitHub account"

**Comment by DeityLink** [comment]
I stand with Exxion's arguments, as for what we're ""losing"" with this PR, well let's review shall we:

- 4 instances of racial slurs in code comments, which means they already don't appear in-game at all.
- 3 item descriptions having a slur removed, that didn't need it in the first place. (for real, the crab's description is a nice reference to Crab Nicholson's Extreme Sleepover, adding a random slur to it doesn't raise its value unless you're some kind of racist)
- 3 instances of swear word replacement/generic swear words for disabilities that have dozen other entries in their list
- 1 less random Tajaran name (there are like 30 others)
- 2 less keys for the gay religion (the 8 others are more than enough) **OH WAIT these actually don't even matter thanks to the new interface we have now to pick a religion**
- 1 less slur in a random PDA news that nobody even reads
- an actual bugfix since turning you into a tajaran is admittedly turning you into a furry, but not necessarily a gay one
- and the **emagged** alarm from the sonic law breaking prevention tool, which admittedly is just **a text version** of beepsky's verbal beatdown, which, guess what, isn't affected by these changes and remains safe as an .ogg file

**Comment by Exxion** [comment]
The Beepsky one is fairly iconic, and I might advocate for putting that one specifically on the secret repo. I'm not convinced we couldn't get away with just leaving it on the main repo, though. It doesn't have any slurs unless you use a particularly broad definition of "slur".

**Comment by ShiftyRail** [comment]
I don't think it's a big deal one way or the other. It all comes down to a matter of pride; are we staying flippant or are we bowing down to the PC overlords? I would be in favour of trimming down the bad words just for visibility. It actually sucks that we have to do it, but we've already gone on the hub and other stuff like that.

Internet changes.

**Comment by Hinaichigo** [comment]
We already went through a whole discussion about this.

**Comment by Hinaichigo** [comment]
gitgud.io

File: code/game/objects/items/devices/PDA/apps/general.dm (hunk 4 above)
Comment (pedr69) [reply code/game/objects/items/devices/PDA/apps/general.dm:145]: this is not in the gamer compliance list

File: code/game/objects/items/robot/robot_items/robot_harm_alarm.dm [hunk 5]
```diff
@@ -65,7 +65,7 @@
 	desc = "Releases a harmless blast that confuses most organics. For when the crime is JUST TOO MUCH"
 	alarm = "HALT! SECURITY!"
 	alarm_sound = 'sound/voice/halt.ogg'
-	emagged_alarm = "FUCK YOUR CUNT YOU SHIT EATING COCKSUCKER MAN EAT A DONG FUCKING ASS RAMMING SHITFUCK. EAT PENISES IN YOUR FUCKFACE AND SHIT OUT ABORTIONS OF FUCK AND DO A SHIT IN YOUR ASS YOU COCK FUCK SHIT MONKEY FUCK ASS WANKER FROM THE DEPTHS OF SHIT."
+	emagged_alarm = "FUCK YOU SHIT EATING Bastard. EAT A DICK AND SHIT OUT ABORTIONS OF FUCK AND DO A SHIT IN YOUR ASS. FUCK SHIT MONKEY FUCK ASS WANKER FROM THE DEPTHS OF SHIT."
```
Comment (SonixApache) [inline code/game/objects/items/robot/robot_items/robot_harm_alarm.dm]:
```suggestion
	emagged_alarm = "FUCK YOU SHIT EATING BASTARD. EAT A DICK AND SHIT OUT ABORTIONS OF FUCK AND DO A SHIT IN YOUR ASS. FUCK SHIT MONKEY FUCK ASS WANKER FROM THE DEPTHS OF SHIT."
```

**Comment by DorainGrey** [comment]
> The Beepsky one is fairly iconic, and I might advocate for putting that one specifically on the secret repo. I'm not convinced we couldn't get away with just leaving it on the main repo, though. It doesn't have any slurs unless you use a particularly broad definition of "slur".

I'm happy to do a define for that one if someone can setup it up on other side.

Only reason I touched it, was they specifically referenced it in email. I guess because the combination of words offended them. If we are going to do this, I'd like to be reasonably certain we will get restored, so I don't want to ignore anything.

As for general defines, I'm not going to touch them. Are people genuinely suggesting they'd change their vote if these were all hidden behind defines? The PR that did this and was downvoted shows otherwise.


At the end of the day, I think it's pretty disingenuous to suggest that the GitHub being hidden behind an account requirement and not even showing up in search is a good thing for the server. For the sake of a few words really, let's be honest for a second. Someone could stealth remove these from server and no one would notice.

**Comment by SonixApache** [comment]
> The Beepsky one is fairly iconic, and I might advocate for putting that one specifically on the secret repo. I'm not convinced we couldn't get away with just leaving it on the main repo, though. It doesn't have any slurs unless you use a particularly broad definition of "slur".

Unfortunately that one's directly on github's nono-word email. Probably just one or two words we have to cut, or we can move the whole thing to the secret repo.

**Comment by Exxion** [comment]
>I'm happy to do a define for that one if someone can setup it up on other side.

No define is necessary to do this.

>Unfortunately that one's directly on github's nono-word email.

I know it is, but it's not impossible that we could convince them to allow it.

**Comment by Hinaichigo** [comment]
This is a worse capitulation than the defines was. Prior to the defines PR, a very similar PR to this was also made and closed.
Is the hope that given enough attempts one will somehow make it through?

**Comment by Theophrastus1** [comment]
The beepsky line is my line in the sand. The one HAS to stay because it is ICONIC.

**Comment by pedr69** [comment]
on one hand i'm gonna miss [redacted]man
on the other hand i dont want to sign in

File: code/datums/religions.dm [hunk 6]
```diff
@@ -702,7 +702,7 @@ var/list/all_bible_styles = list(
 	bible_type = /obj/item/weapon/storage/bible/booze
 	male_adept = "LGBT Advocate"
 	female_adept = "LGBT Advocate"
-	keys = list("homosexuality", "faggotry", "gayness", "gay", "penis", "faggot", "cock", "cocks", "dick", "dicks")
+	keys = list("homosexuality", , "gayness", "gay", "penis", "cock", "cocks", "dick", "dicks")
```
Comment (Kurfursten) [inline code/datums/religions.dm]: don't just leave a null entry in the key list!

File: code/datums/gamemode/role/catbeast.dm [hunk 7]
```diff
@@ -38,7 +38,7 @@
 
 var/list/catbeast_names = list("Meowth","Fluffy","Subject 246","Experiment 35a","Nyanners","Thing From Below","Airlock Scratcher","Flees-Like-Fleas",
 						"Lurks-In-Shadows","Eartha Kitt","Target Practice","Fresh Meat","Ca'thulu","Furry Fury","Vore-Strikes-Back","Killing Machine","Uncle Tom",
-						"Nine Lives", "Bad Luck", "Siamese Sam", "Tom Tabby", "Hairball", "Throws-Dice-Poorly", "Wizard Apprentice", "Lynch Lynx", "Felix", "Niggerman")
+						"Nine Lives", "Bad Luck", "Siamese Sam", "Tom Tabby", "Hairball", "Throws-Dice-Poorly", "Wizard Apprentice", "Lynch Lynx", "Felix")
```
Comment (Kurfursten) [inline code/datums/gamemode/role/catbeast.dm]: Someone suggested "Lovecraft's Cat" could replace this, which I think was funny. I wrote every catbeast name except this one in question (which Adam submitted)

**Comment by PrimeDSS13** [comment]
what does the beepsky line actually change in game?
correct me if I'm wrong but the audio file should remain the same, right? in this instance, what does the changed code actually affect about it?

**Comment by nervere** [comment]
> what does the beepsky line actually change in game? correct me if I'm wrong but the audio file should remain the same, right? in this instance, what does the changed code actually affect about it?

It prints to chat when you use an emagged harm alarm. It doesn't affect beepsky, it's just a reference to the sound file.

**Comment by PrimeDSS13** [comment]
> > what does the beepsky line actually change in game? correct me if I'm wrong but the audio file should remain the same, right? in this instance, what does the changed code actually affect about it?
> 
> It prints to chat when you use an emagged harm alarm. It doesn't affect beepsky, it's just a reference to the sound file.

that's what I assumed it would do really, though I didn't know that any harm alarm could perform it
realistically I'd rather it just not print a message to chat instead of it printing a bastardized version of the classic instead

**Comment by DeityLink** [comment]
> The beepsky line is my line in the sand. The one HAS to stay because it is ICONIC.

It stays, that's not what this PR is removing. Here it's just the emagged hailer's line.

**Comment by Theophrastus1** [comment]
> > The beepsky line is my line in the sand. The one HAS to stay because it is ICONIC.
> 
> It stays, that's not what this PR is removing. Here it's just the emagged hailer's line.

emagged hailer is one of my favorite items man. i will not bend the knee to lose these two things.

**Comment by Exxion** [comment]
If necessary, I'll recreate that specific one on the secret repo

**Comment by calzilla1** [comment]
Personally I'm just glad we can finally have the "nigger" religion without discrimination

File: code/datums/gamemode/role/catbeast.dm (hunk 7 above)
Comment (SonixApache) [reply code/datums/gamemode/role/catbeast.dm]: i'll have you know that i came up with nyanners

**Comment by Inorien** [comment]
kinda wish github wouldve just closed the repo without warning, that would have been based

**Comment by DeityLink** [comment]
totally

File: code/datums/gamemode/role/catbeast.dm (hunk 7 above)
Comment (DorainGrey) [reply code/datums/gamemode/role/catbeast.dm]: Based Idea, did it there.

**Comment by DorainGrey** [comment]
Updated based on Suggestions, I will leave the Beepsky line alone as I believe Exxion will overwrite that in the secret repo alone. Using the framework from this PR #34407

**Comment by Exxion** [comment]
Yeah, that PR lets it be overridden in the secret repo without anything special needed in this repo (besides that PR)
It's also useful for other things, and I've been meaning to make that PR for quite a while

**Comment by BomberBro** [comment]
The code comments can be killed because, frankly, who cares. Everything that you can see in game should be thrown in the secret repo though, including lovecraft's cat.

**Comment by Theophrastus1** [comment]
i have thumbs upped so long as we retain the emagged hailer and beepsky.

**Comment by tinysaturn** [comment]
the repo being hidden is so inconsequential
The average player doesn't give a shit about some github repo and will never look at it anyway
The player that wants to get engaged in discussion about the PRs will have already made an account, nobody actually gives a shit about anonymous, no-name-attached opinions on PRs (search your heart you know it's true).
The coder already has an account, because he has to have one, so he doesn't give a shit about the repo being hidden

**Comment by pedr69** [comment]
what about the average player who sees a big "new server poll" button which then links him to a "create shithub to continue"

**Comment by ShiftyRail** [comment]
Not like I enjoy doing it, but y'know, it is what it is.

**Comment by Hinaichigo** [comment]
only in ohio 💀

**Comment by ShiftyRail** [comment]
image

I'm not putting back the Lovecraft's cat though

**Comment by xeroxemnas** [comment]
Shifty trying hard to emulate his hero Macron I see.

**Comment by wotisjanitor** [comment]
image

**Comment by ShiftyRail** [comment]
>putting aside the absurd idea that you have to argue against something rather than for it, there are absolutely reasons someone might not like this

Correct. That person might be an Hitleritr

**Comment by John-VI** [comment]
Repo is still hidden. Have you considered making a burnt offering?

**Comment by pedr69** [comment]
@d3athrow please let github know that we are now gamer compliant

**Comment by optimumtact** [comment]
coolguy

**Pull request merged**